feat(providers): add Shopify provider (#3964)
nextauthjs/next-auth: docs/docs/providers/shopify.md

---
id: shopify
title: Shopify
---

## Documentation

https://shopify.dev/apps/auth/oauth/getting-started

## Configuration

> This is your **Redirect URL** <ins>http://localhost:3000/api/auth/callback/shopify</ins>. Add this callback URL in your app settings.

Reviewer: I would re-word this. 

`http://localhost:3000/api/auth/callback` works if the user is trying the application locally, however in other environments like production the domain is going to be different. 

For instance, when using **Vercel** it would be: `https://my-app.vercel.app/api/auth/callback`.

```suggestion
Add `/api/auth/callback/shopify in your app settings as the **callback URL**. For instance, if you're trying the sign in locally it'll be: `http://localhost:3000/api/auth/callback`.
```

:::tip
This guide doesn't apply if you created a custom app in the Shopify admin. Refer to [Access tokens for custom apps in the Shopify admin](https://shopify.dev/apps/auth/admin-app-access-tokens).

If you're building an embedded app, then refer to our tutorial on [authenticating an embedded app using OAuth and session tokens](https://shopify.dev/apps/auth/oauth/session-tokens/getting-started).
:::

## Options

The **Shopify Provider** comes with a set of default options:

- [Shopify Provider options](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/shopify.js)

You can override any of the options to suit your own use case.

## Example

```js
import ShopifyProvider from "next-auth/providers/shopify";
...
providers: [
  ShopifyProvider({
    clientId: process.env.SHOPIFY_ID,
    clientSecret: process.env.SHOPIFY_SECRET,
    shop: "unique-master", // Don't include .myshopify.com

>     scope: "read_products,write_products,read_orders,write_orders",

Reviewer: ```suggestion
    authorization: {
      // if needed, you can ask for extra scopes here...
      params: { scope: "read_products" },
    },
```

> ---

Reviewer 2: Let's make sure that this documentation page has the same structure as others, with the following headings:

- Documentation
- Configuration
- Options
- Example

id: shopify
title: Shopify
---

> ## Documentation for Shopify apps
> https://shopify.dev/apps
> 
> 

Reviewer 2: ```suggestion
## Documentation for Shopify apps

https://shopify.dev/apps

```

## Options

The **Shopify Provider** comes with a set of default options:

- [Shopify Provider options](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/shopify.js)

You can override any of the options to suit your own use case.

## Example

```js
import ShopifyProvider from "next-auth/providers/shopify";
...
providers: [
  ShopifyProvider({
    clientId: process.env.SHOPIFY_ID,
    clientSecret: process.env.SHOPIFY_SECRET,

>     shop: "unique-master", // Don't include .myshopify.com
>     scope: "read_products,write_products,read_orders,write_orders",
>     apiVersion: "2022-01",

Reviewer 2: ```suggestion
```

  })
]
...
```

## Instructions

### Configuration

:::tip
Here is the docs for Shopify Oauth https://shopify.dev/apps/auth/oauth/getting-started
:::

:::tip
Create your shopify partner account https://www.shopify.com/shopifypartnersvap and create an app from there

Under "URLs" in the App Setup, configure the following for "OAuth 2.0":

- Redirect URL
  - http://localhost:3000/api/auth/callback/shopify

> :::

Reviewer 2: Let's use `tip` with moderation. Also, keep the casings consistent.

OAuth, Shopify,
